workers/ohsome_quality_analyst/ohsome/client.py

```diff
@@ -20,52 +20,76 @@ async def query(
     ratio: bool = False,
 ) -> Optional[Dict]:
     """Query ohsome API endpoint with filter."""
+    url = _build_url(layer, endpoint, ratio)
+    data = _build_data_dict(layer, bpolys, time, ratio)
+    logging.info("Query ohsome API.")
+    logging.debug("Query URL: " + url)
+    logging.debug("Query filter: " + layer.filter)
+    logging.debug("Query data: " + json.dumps(data))
+    return await _query_ohsome_api(url, data)
+
+
+async def get_latest_ohsome_timestamp():
+    """Get unix timestamp of ohsome from ohsome api."""
+    url = "https://api.ohsome.org/v1/metadata"
+    async with httpx.AsyncClient() as client:
+        resp = await client.get(url)
+    timestamp_str = str(resp.json()["extractRegion"]["temporalExtent"]["toTimestamp"])
+    timestamp = datetime.datetime.strptime(timestamp_str, "%Y-%m-%dT%H:%MZ")
+    return timestamp
+
+
+def _build_url(
+    layer,
+    endpoint: Optional[str] = None,
+    ratio: bool = False,
+) -> str:
+    """Build endpoint URL of ohsome API."""
+    ohsome_api = OHSOME_API.removesuffix("/")
     if endpoint is not None:
-        url = OHSOME_API + endpoint
+        url = ohsome_api + "/" + endpoint.removesuffix("/")
     else:
-        url = OHSOME_API + layer.endpoint
+        url = ohsome_api + layer.endpoint.removesuffix("/")
+    if ratio:
+        return url + "/" + "ratio"
+    return url
 
+
+def _build_data_dict(
+    layer,
+    bpolys: str,
+    time: Optional[str] = None,
+    ratio: bool = False,
+) -> dict:
+    """Build data dictionary for ohsome API query."""
     if ratio:
-        url = url + "/ratio"
         data = {"bpolys": bpolys, "filter": layer.filter, "filter2": layer.ratio_filter}
     else:
         data = {"bpolys": bpolys, "filter": layer.filter}
 
     if time is not None:
         data["time"] = time
+    return data
 
-    logging.info("Query ohsome API.")
-    logging.debug("Query data: " + json.dumps(data, indent=4, sort_keys=True))
 
+async def _query_ohsome_api(url: str, data: dict):
     # set custom timeout as ohsome API takes a long time to send an answer
     timeout = httpx.Timeout(5, read=600)
```

Comment:
Make timeout a bit longer than the timeout of ohsome API. Invalid response of ohsome API due to timeout is catched later in the code.